### PR TITLE
Add cucumber scenarios

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: cristin-import/src/test/resources/features/GeneralMappingRules.feature [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    And the CristinTitle array has an entry
+    And the CristinTitle entry has title text equal to "This is some title"
+    And the CristinTitle entry has original status annotation "J"
+    When the Cristin Result is converted to an NVA Resource
+    Then the NVA Resource has an EntityDescription with mainTitle "This is some title"
+
+
+  Scenario: map returns NVA Resource with main title being the Cristin title annotated as
+  Original Title when there are many titles but only one annotated as original
+    Given the Cristin Result has an non null array of CristinTitles
```
Comment (brinxmat) [inline cristin-import/src/test/resources/features/GeneralMappingRules.feature:23]: I'd evaluate using the following pattern in this and the following scenarios

```
    Given the Cristin Result has an array of CristinTitles with values:
      | TitleText                    | OriginalStatus |
      | This is the original title   | J              |
      | This is the translated title | N              |
```

File: cristin-import/src/test/resources/features/GeneralMappingRules.feature [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    And the CristinTitle entry has original status annotation "J"
+    And the CristinTitle array has an entry
+    And the CristinTitle entry has title text equal to "This is translated title"
+    And the CristinTitle entry has original status annotation "N"
+    When the Cristin Result is converted to an NVA Resource
+    Then the NVA Resource has an EntityDescription with mainTitle "This is the original title"
+
+
+  Scenario: map returns NVA Resource with Main Title being any Cristin Title annotated as
+  Original Title when there are two titles both only one annotated as original
```
Comment (brinxmat) [inline cristin-import/src/test/resources/features/GeneralMappingRules.feature:35]: Is there an error here as both are annotated as original

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java [hunk 3]
```diff
@@ -192,11 +192,11 @@ private PublicationDate extractPublicationDate() {
     }
 
     private String extractMainTitle() {
-        return extractCristinTitles()
-                   .filter(CristinTitle::isMainTitle)
-                   .findFirst()
-                   .map(CristinTitle::getTitle)
-                   .orElseThrow();
+        var x = extractCristinTitles()
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:195]: I would make a remark, but x marks the spot

File: cristin-import/src/test/java/cucumber/DataTableTypeTransformations.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package cucumber;
+
+import io.cucumber.java.DataTableType;
+import java.util.Map;
+import no.unit.nva.cristin.mapper.CristinTitle;
+
+public class DataTableTypeTransformations {
+
+    public static final int CURRENTLY_MAPPED_FIELDS = 2;
+    public static final String WRONG_NUMBER_OF_FIELDS_FOR_CRISTINT_TITLE =
```
Comment (brinxmat) [inline cristin-import/src/test/java/cucumber/DataTableTypeTransformations.java:10]: Typo in CRISTINT_TITLE, CHRIST_INT_TITLE

File: cristin-import/src/test/resources/features/GeneralMappingRules.feature [hunk 5]
```diff
@@ -11,35 +11,30 @@ Feature: Mappings that hold for all types of Cristin Results
   Scenario: map returns NVA Resource with main title being the Cristin title annotated as
   Original Title when there is only one CristinTitle and it is annotated as original
     Given the Cristin Result has an non null array of CristinTitles
-    And the CristinTitle array has an entry
-    And the CristinTitle entry has title text equal to "This is some title"
-    And the CristinTitle entry has original status annotation "J"
+    Given the Cristin Result has an  CristinTitles with values:
```
Comment (brinxmat) [inline cristin-import/src/test/resources/features/GeneralMappingRules.feature:14]: has a CristinTitles

Comment (brinxmat) [reply cristin-import/src/test/resources/features/GeneralMappingRules.feature:14]: Also double space there

Comment (brinxmat) [reply cristin-import/src/test/resources/features/GeneralMappingRules.feature:14]: Applies for every case

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java (hunk 3 above)
Comment (axthosarouris) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:195]: woops :1st_place_medal:

**Review by brinxmat — APPROVED** [review]

**Pull request merged**